src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "./StrategyBase.sol";
+import "../permissions/Pausable.sol";
+
+/**
+ * @title TODO: write this
+ * @author Layr Labs, Inc.
+ * @notice TODO: write this
+ */
+contract StrategyFactory is OwnableUpgradeable, Pausable {
+
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    StrategyBase public immutable strategyImplementation;
+
+    ProxyAdmin public eigenLayerProxyAdmin;
+
+    // @notice Mapping token => strategy contract for the token
+    mapping(IERC20 => IStrategy) public tokenStrategies;
+
+    event ProxyAdminChanged(ProxyAdmin previousProxyAdmin, ProxyAdmin newProxyAdmin);
+    event StrategySetForToken(IERC20 token, IStrategy strategy);
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager, StrategyBase _strategyImplementation) {
+        strategyManager = _strategyManager;
+        strategyImplementation = _strategyImplementation;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        ProxyAdmin _eigenLayerProxyAdmin
+    )
+        public virtual initializer
+    {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        // TODO: decide if a function for changing this is warranted
+        eigenLayerProxyAdmin = _eigenLayerProxyAdmin;
+        emit ProxyAdminChanged(ProxyAdmin(address(0)), _eigenLayerProxyAdmin);
+    }
+
+    // TODO: document with ample warnings
+    function deployNewStrategy(IERC20 token) external onlyWhenNotPaused(PAUSED_NEW_STRATEGIES) {
+        require(tokenStrategies[token] == IStrategy(address(0)),
+            "StrategyFactory.deployNewStrategy: Strategy already exists for token");
+        IStrategy strategy = IStrategy(
+            address(
+                new TransparentUpgradeableProxy(
+                    address(strategyImplementation),
+                    address(eigenLayerProxyAdmin),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, token, pauserRegistry)
+                )
+            )
+        );
+        _setStrategyForToken(token, strategy);
+    }
+
+    /** 
+     * @notice Owner-only function to pass through a call to `StrategyManager.addStrategiesToDepositWhitelist`
+     * @dev Also adds the `strategiesToWhitelist` to the `tokenStrategies` mapping
+     */
+    function whitelistStrategies(
```

Comment:
shouldn't we just accept an array of tokens here instead of strategies and the strategy is looked up in the `tokenStrategies` mapping?

Reply:
what? super confused on what you think the intended usage is

Reply:
looks like I had missed perhaps the most important function call to have in the `deployNewStrategy` function. added here; please resolve this thread if this makes sense now.
https://github.com/Layr-Labs/eigenlayer-contracts/pull/522/commits/0e572603d34df67a65b32c8a31f4d59f6774fe5b